discussment-ui-wicket/src/main/java/org/danekja/discussment/ui/wicket/form/TopicForm.java

```diff
@@ -61,16 +67,18 @@ public void setTopicService(TopicService topicService) {
 
     @Override
     protected void onSubmit() {
-        if (topicService != null) {
-            try {
-                topicService.createTopic(categoryModel.getObject(), topicModel.getObject());
-            } catch (AccessDeniedException e) {
-                // todo: not yet impemented
+        try {
+            Topic topic = topicService.createTopic(categoryModel.getObject(), topicModel.getObject());
+
+            if (!accessControlService.canAddDiscussion(topic)) {
+                permissionService.configureDiscussionPermissions(userService.getCurrentlyLoggedUser(), topic, new PermissionData(true, false, false, true));
             }
 
-            topicModel.setObject(new Topic());
-            setResponsePage(getPage().getPageClass(), getPage().getPageParameters());
+        } catch (AccessDeniedException e) {
+            // todo: not yet impemented
```

Comment:
Todo?

Reply:
Well, the same todo is everywhere `AccessDeniedException` is caught. It was not my focus in this PR.